File: haskell/src/Solutions/Euler146.hs
module Solutions.Euler146 where
import Math.NumberTheory.Primes.Testing(isPrime)

check n =
    let f = map ((n * n)+)
    in all isPrime (f [1, 3, 7, 9, 13, 27]) && not (any isPrime $ f [11, 17, 19, 21, 23])
f n = sum $ filter check [10, 20..n - 1]

euler146 :: IO String
euler146 = return $ show $ f 150000000

File: rust/src/solutions/euler146.rs
use primal::is_prime;

fn check(&n: &u64) -> bool {
    let f = |x| n * n + x;
    [1, 3, 7, 9, 13, 27].iter().all(|&x| is_prime(f(x)))
        && ![11, 17, 19, 21, 23].iter().any(|&x| is_prime(f(x)))
}

pub fn euler146() -> String {
    (10..150000000)
        .step_by(10)
        .filter(check)
        .sum::<u64>()
        .to_string()
}
